lib/spack/spack/cmd/config.py

```diff
@@ -131,32 +133,50 @@ def _get_scope_and_section(args):
     return scope, section
 
 
+def print_configuration(args, *, blame: bool) -> None:
```

Comment:
Can this stuff be pushed into `spack.config`?  I think it makes a lot of sense to have the `Configuration` be constructed with the top-level env key, and to make the `path` arg of allow spack.config.get() optional.  Thoughts?

Reply:
That sounds fine to me. Pushing to `Configuration` will likely be a larger change than the current PR, due to the need to refactor methods / functions where we just hacked the environment scope in configuration, by treating it ad-hoc.

Reply:
Ok let's merge this as-is for now and save that refactor for another PR. We will need that refactor for pulling config out into an external package, but since this is useful now, let's put this in and do the refactor after.